### PR TITLE
Fixing up Mock API tests post C# Beta.

### PR DESCRIPTION
**Pull request by rzaracota**
Fixes an issue with errors being wrapped by the async task

### CODE REVIEW

**Comment by timothodge** [comment]
Fixed up the PR to deserialize RestSharp responses, unwrap AggregateExceptions, and help pass back correct error code/messages for mock API tests that are testing error paths.

**Review by timothodge — COMMENTED** [review]
I approve of these changes but I would appreciate @lifebeyondfife or @pratik136 to also take a look before we move forward.

**Review by pratik136 — DISMISSED** [review]
The changes look good to me. 🎉

**Review by lifebeyondfife — DISMISSED** [review]
Looks good 😄

**Review by timothodge — DISMISSED** [review]

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -26,15 +26,15 @@ jobs:
...
-    #    smartsheet-sdk-tests/travis_scripts/start_wiremock.sh
-    #- name: Run Mock API Tests
-    #  working-directory: ./mock-api-test-sdk-net60
-    #  run: |
-    #    dotnet build --configuration Release --no-restore
-    #    dotnet test --no-restore
+    - name: Setup Mock API
+      run: |
+       smartsheet-sdk-tests/travis_scripts/install_wiremock.sh
+       smartsheet-sdk-tests/travis_scripts/start_wiremock.sh
```
Comment (pratik136) [inline .github/workflows/publish.yml]: Referencing the old path `.../travis_scripts/...`. 

Change to 
```sh
    smartsheet-sdk-tests/ci_scripts/install_wiremock.sh
    smartsheet-sdk-tests/ci_scripts/start_wiremock.sh
```

**Review by pratik136 — CHANGES_REQUESTED** [review]

**Review by pratik136 — APPROVED** [review]

**Pull request merged**